### PR TITLE
INT-172 : Add support for POST /referrals endpoint

### PR DESCRIPTION
**Pull request by muhammadatdoshii**
[INT-172](https://doshii.atlassian.net/browse/INT-172)

### CODE REVIEW

File: lib/referral.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { AxiosRequestConfig } from "axios";
+
+export enum ReferralStatus {
+  PENDING = "pending",
+  REJECTED = "rejected",
+  ACCEPTED = "accepted",
+  COMPLETE = "complete",
+  CANCELLED = "cancelled"
+}
```
Comment (johnatdoshii) [inline lib/referral.ts:9]: The only status that I am aware of that is supported is `pending` (and based on the ReferralFixture in the Partner API, it is the only one it supports). The rest of the status have not yet been defined by the business - they are currently working on the changes to Connect to define these.

**Review by johnatdoshii — CHANGES_REQUESTED** [review]

File: lib/referral.ts (hunk 1 above)
Comment (johnatdoshii) [reply lib/referral.ts:9]: At the moment, we don't know what the other status will be, so we'll need to only support `pending` or we put this PR on hold until Connect complete their analysis (a couple of weeks I would expect). 🤔

Comment (muhammadatdoshii) [reply lib/referral.ts:9]: Alright, I'll leave this on hold until Connect decide on the statuses

File: lib/referral.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export interface ReferralRequest {
+  email: string;
+  firstName?: string;
+  lastName?: string;
+  phone?: string;
+  venueName?: string;
+  partnerLocationId?: string;
+  doshiiLocationId?: string;
+  doshiiOrganisationId?: string;
+  address: {
```
Comment (carlosatdoshii) [inline lib/referral.ts:20]: address should be optional, as we cannot guaranty CBA aways send the address for us during the onboarding process.

Comment (johnatdoshii) [reply lib/referral.ts:20]: We should check this schema against the Joi schema defined in the PartnerController (in Partner API repo)

**Comment by sverreatdoshii** [comment]
Closing after discussion with @carlosatdoshii

**Pull request closed without merging**